src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,158 @@
+        private final K key;
+        private V value;
+        private Entry<K, V> next;
+
+        Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            Entry<?, ?> entry = (Entry<?, ?>) o;
+
+            return Objects.equals(key, entry.key);
+        }
+
+        @Override
+        public int hashCode() {
+            return key != null ? key.hashCode() : 0;
+        }
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if ((size + 1) > (capacity * LOAD_FACTOR)) {
+            resize();
+        }
+        int hash = getIndexByTheKey(key);
+        Entry<K, V> e = table[hash];
+        if (e == null) {
+            table[hash] = new Entry<>(key, value);
+            size++;
+        } else {
+            while (e.next != null) {
+                if ((e.key != null) && (e.key.equals(key))) {
+                    e.value = value;
+                    return;
+                }
+                if (e.key == null && key == null) {
+                    e.value = value;
+                    return;
+                }
+                e = e.next;
+            }
+            if ((e.key != null) && (e.key.equals(key))) {
+                e.value = value;
+                return;
+            }
+            if (e.key == null && key == null) {
+                e.value = value;
+                return;
+            }
+            e.next = new Entry<>(key, value);
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int hash = getIndexByTheKey(key);
+        Entry<K, V> e = table[hash];
+        if (e == null) {
+            return null;
+        }
+        while (e != null) {
+            if (key == null) {
+                if (e.key == null) {
+                    return e.value;
+                }
+            } else if (key.equals(e.key)) {
+                return e.value;
+            }
+            e = e.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void resize() {
+        Entry<K, V>[] newTable = new Entry[capacity * 2];
+        for (Entry<K, V> e : table) {
+            if (e != null) {
+                while (e.next != null) {
+                    transfer(e.key, e.value, newTable);
+                    e = e.next;
+                }
+                transfer(e.key, e.value, newTable);
+            }
+        }
+        table = newTable;
+        capacity *= 2;
+    }
+
+    private void transfer(K key, V value, Entry<K, V>[] newTable) {
+        int hash = (key == null) ? 0 : (key.hashCode() % (capacity * 2));
```

Comment:
You can change capacity before calculating hash, and move this logic(for hash) in a separate method, because you use it twice